src/Amqp/JobQueue/AmqpQueue.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Icecave\Overpass\Amqp\JobQueue;
+
+use Icecave\Overpass\JobQueue\Message\MessageSerialization;
+use Icecave\Overpass\JobQueue\Message\MessageSerializationInterface;
+use Icecave\Overpass\JobQueue\Message\Request;
+use Icecave\Overpass\JobQueue\QueueInterface;
+use Icecave\Overpass\Serialization\JsonSerialization;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerAwareTrait;
+
+class AmqpQueue implements QueueInterface
+{
+    use LoggerAwareTrait;
+
+    /**
+     * @param AMQPChannel                        $channel
+     * @param DeclarationManager|null            $declarationManager
+     * @param MessageSerializationInterface|null $serialization
+     */
+    public function __construct(
+        AMQPChannel $channel,
+        DeclarationManager $declarationManager = null,
+        MessageSerializationInterface $serialization = null
+    ) {
+        $this->channel = $channel;
+        $this->declarationManager = $declarationManager ?: new DeclarationManager($channel);
+        $this->serialization = $serialization ?: new MessageSerialization(new JsonSerialization());
+    }
+
+    /**
+     * Add a task to the job queue.
+     *
+     * @param string $task    The name of the task to enqueue
+     * @param mixed  $payload The task payload to pass
+     */
+    public function enqueue($task, $payload = null)
+    {
+        $request = Request::create($task, $payload);
+
+        if ($this->logger) {
+            $this->logger->debug(
+                'jobqueue.queue enqueue successful: {request}',
+                [
+                    'request' => $request,
+                ]
+            );
+        }
+
+        $this->send($request);
+    }
+
+    /**
+     * Add a task to the job queue.
+     *
+     * @param string $task      The name of the task to enqueue
+     * @param array  $arguments The arguments to pass to enqueue
+     */
+    public function __call($task, array $arguments)
```

Comment:
Ditch `__call()` - it only really makes sense in the RPC system, and even there we never use it because the procedure names aren't valid PHP method names anyway.